src/symbol_object.cpp

```diff
@@ -101,9 +145,10 @@ Value SymbolObject::name(Env *env) const {
     }
     return symbol->m_string;
 }
-
-Value SymbolObject::ref(Env *env, Value index_obj) {
-    return to_s(env)->send(env, intern("[]"), { index_obj });
+Value SymbolObject::ref(Env *env, Value index_obj, Value length_obj) {
+    // The next line worked in nearly every case, except it did not set `$~`
+    //return to_s(env)->send(env, intern("[]"), { index_obj, length_obj });
```

Comment:
I poked around for a while but never could figure out why `$~` (last match) did not get set by sending the `[]` method, but directly using `to_s(env)->ref(...)` worked okay.

Reply:
I'm not sure, but we can revisit later.